sdk/src/main/scala/coop/rchain/sdk/dag/merging/TipConflictResolver.scala

```diff
@@ -0,0 +1,69 @@
+package coop.rchain.sdk.dag.merging
+import coop.rchain.sdk.dag.merging.ConflictResolutionLogic.removeDependentChains
+
+/**
+  * Conflict resolver aiming merging as less deploys as possible.
+  * Tip with the most deploys from the conflict scope in the view (base tip) is ued as a base for the merge.
+  * This resolver can only be used if finalization does not alter the state of a base tip.
+  */
+object TipConflictResolver {
```

Comment:
_Tip_ block doesn't seems to be a good naming here. Conflict detection can be done (and it is in validation) on any set of blocks that are merged, of course in some moment they will be tip of the DAG.

Maybe `ConcurrentConflictResolver` of `SiblingsConflictResolver` can be a better fitted name?

Reply:
`(and it is in validation)` could you point me to the code? I'm not sure I understand wdy mean.

Tip here means that some tip (which is a block from latest messages, so its really a tip) is a base state.

Reply:
Conflict resolver works for any blocks in the DAG, not just blocks which are currently tip of the DAG. Every block is created from _tips_ in that moment, but when this block is replayed the same blocks doesn't have to be tips anymore because new blocks are created on top of them.

Reply:
I'm not sure I follow - conflict resolver does not care and does not know about anything what is on top of latest messages. So from the point of conflict resolution it is a tip.

Conflict resolver (CR) inputs latests messages, these are tips according to CR. And the purpose is to find a way to merge views of these tips. So you can resolve conflicts by merging into some state which is a fringe below tips, or into one of the tips. 

Maybe `TipConflictResolver ` is a bad name - what It supposed to mean that it produces conflict resolution to merge into one of latest messages supplied. Or do you mean we should not call latest messages with `tips`?

Reply:
If we take this DAG as example.

Is it possible to use blocks `0-4` and `1-4` to resolve conflicts? Are these blocks tips of the DAG? 

We can ask the same question for blocks `0-4` and `3-5`.

So to understand the code we need to know the meaning of _tips_. We are considering `0-4`, `3-5` as part of tips because they are created from different validator but if they are in sequence conflict detection is meaningless.
And in this DAG these blocks are not tips, but we can still calculate conflicts.

So what I'm suggesting is a name that uses less assumptions about the meaning and more precisely describe the purpose.

![image](https://user-images.githubusercontent.com/5306205/175508220-577d2952-1216-41c7-8adf-3e9045b1a0f8.png)

Reply:
If you submit `0-4` and `1-4` as latest messages, and `0-1` `1-2` `2-2` `3-2` as a fringe - I think conflict resolution logic and merging still can and should work fine. Conflict set is what is seen in latest messages and not seen by the fringe. You need to merge this conflict set, and I don't see why you can not. And you have options to merge into fringe you provided, or into one of tips you provided.

Reply:
`any concurrent combination` - how these can be not tips? if you merged these concurrent blocks - these are the topmost ones in the view you are merging, so tips.

Reply:
What I want to point out is that tips of the DAG can be all blocks from every validator and we can detect conflicts between just two blocks which can be subset of _tips_.

I have a problem with the meaning of _tips_. Can you find some reference where this term is use in the context of DAGs or graphs in general?

Reply:
`we can have regular situation in replay when we pick blocks from the DAG not seen as latest`
I don't get this - when we replay block we have a view of the block, where blocks we are talking about **are** latest/tips.

Also - its fine to have tips in sequence. If from tips {a,b} only `b` is not altered by rejections due to finalization, and `a` sees `b` but altered, it's still fine to have the state of `b` as a base for merge.

In general case we can pick any block in the conflict set as a base for merge, but this particular one is merging only into tips - one of the latest messages from all senders.